bin/cron.js

```diff
@@ -100,16 +113,55 @@ cron.schedule("*/15 * * * *", async () => {
           if (userProgress) {
             user.submissions.map(async (submission) => {
               const moduleID = assignmentIdToType[submission.assignment_id].moduleID; // Map current submission id to moduleID
-              logs.success[user.user_id] = { practice: [], apply: [], daily: [], reflection: [] };
+              const subj = assignmentIdToType[submission.assignment_id].subject;
 
+              logs.success[user.user_id] = {
+                practice: [],
+                apply: [],
+                daily: [],
+                reflection: [],
+                econ: [],
+                bio: [],
+                chem: [],
+                physics: [],
+                engineering: [],
+              };
               switch (assignmentIdToType[submission.assignment_id].type) {
                 case "practice":
                   // Hard-coded to 90 for now
                   if (submission.score >= 90) {
                     score += 100;
                     completed.practice += 1;
                     // If submission not already stored in MongoDB
-                    if (!userProgress.modules || !(moduleID in userProgress.modules)) {
+                    if (subj) {
+                      switch (subj) {
+                        case "econ":
+                          completed.econ += 1;
+                          logs.success[user.user_id].econ.push(submission.assignment_id);
+                          break;
+                        case "bio":
+                          completed.bio += 1;
+                          logs.success[user.user_id].bio.push(submission.assignment_id);
+                          break;
+                        case "chem":
+                          completed.chem += 1;
+                          logs.success[user.user_id].chem.push(submission.assignment_id);
+                          break;
+                        case "physics":
+                          completed.physics += 1;
+                          logs.success[user.user_id].physics.push(submission.assignment_id);
+                          break;
+                        case "engineering":
+                          completed.engineering += 1;
+                          logs.success[user.user_id].engineering.push(submission.assignment_id);
+                          break;
+                      }
+                    }
```

Comment:
This looks good! You can probably make this shorter by just doing `completed[subj] += 1` instead of all the switch-cases